src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerFeature.cs

```diff
@@ -2,18 +2,26 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
 
 namespace Microsoft.AspNetCore.Diagnostics
 {
     /// <summary>
     /// A feature containing the path and error of the original request for examination by an exception handler.
     /// </summary>
-    public class ExceptionHandlerFeature : IExceptionHandlerPathFeature
+    public class ExceptionHandlerFeature : IExceptionHandlerEndpointFeature
     {
         /// <inheritdoc/>
         public Exception Error { get; set; } = default!;
 
         /// <inheritdoc/>
         public string Path { get; set; } = default!;
+
+        /// <inheritdoc/>
+        public Endpoint? Endpoint { get; set; } = default!;
```

Comment:
You shouldn't need to set a default since these properties are nullable.
```suggestion
        public Endpoint? Endpoint { get; set; }
```